src/main/java/org/graylog/integrations/aws/codecs/KinesisCloudWatchFlowLogCodec.java

```diff
@@ -0,0 +1,115 @@
+import org.graylog2.plugin.configuration.Configuration;
+import org.graylog2.plugin.configuration.ConfigurationRequest;
+import org.graylog2.plugin.inputs.annotations.ConfigClass;
+import org.graylog2.plugin.inputs.annotations.FactoryClass;
+import org.graylog2.plugin.inputs.codecs.AbstractCodec;
+import org.graylog2.plugin.inputs.codecs.Codec;
+import org.joda.time.Seconds;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import java.util.HashMap;
+import java.util.Map;
+
+public class KinesisCloudWatchFlowLogCodec extends AbstractKinesisCodec {
+    public static final String NAME = "FlowLog";
+
+    private final IANAProtocolNumbers protocolNumbers;
+
+    @Inject
+    public KinesisCloudWatchFlowLogCodec(@Assisted Configuration configuration, ObjectMapper objectMapper) {
+        super(configuration, objectMapper);
+        this.protocolNumbers = new IANAProtocolNumbers();
+    }
+
+    @Nullable
+    @Override
+    public Message decodeLogData(@Nonnull final KinesisLogEntry logEvent) {
+        try {
+            final FlowLogMessage flowLogMessage = FlowLogMessage.fromLogEvent(logEvent);
+
+            if (flowLogMessage == null) {
+                return null;
+            }
+
+            final String source = configuration.getString(KinesisCloudWatchFlowLogCodec.Config.CK_OVERRIDE_SOURCE, "aws-flowlogs");
+            final Message result = new Message(
+                    buildSummary(flowLogMessage),
+                    source,
+                    flowLogMessage.getTimestamp() );
+            result.addFields(buildFields(flowLogMessage));
+            result.addField(FIELD_KINESIS_STREAM, logEvent.kinesisStream());
+            result.addField(FIELD_LOG_GROUP, logEvent.logGroup());
+            result.addField(FIELD_LOG_STREAM, logEvent.logStream());
+            result.addField(SOURCE_GROUP_IDENTIFIER, true);
+
+            return result;
+        } catch (Exception e) {
+            throw new RuntimeException("Could not deserialize AWS FlowLog record.", e);
+        }
+    }
+
+    private String buildSummary(FlowLogMessage msg) {
+        return new StringBuilder()
+                .append(msg.getInterfaceId()).append(" ")
+                .append(msg.getAction()).append(" ")
+                .append(protocolNumbers.lookup(msg.getProtocolNumber())).append(" ")
+                .append(msg.getSourceAddress()).append(":").append(msg.getSourcePort())
+                .append(" -> ")
+                .append(msg.getDestinationAddress()).append(":").append(msg.getDestinationPort())
+                .toString();
+    }
+
+    private Map<String, Object> buildFields(FlowLogMessage msg) {
+        return new HashMap<String, Object>() {{
+            put("account_id", msg.getAccountId());
+            put("interface_id", msg.getInterfaceId());
+            put("src_addr", msg.getSourceAddress());
+            put("dst_addr", msg.getDestinationAddress());
```

Comment:
From @kroepke:
> i think it would be good to have an optional prefixing in there
> that's done in the beats plugin as well
> by default i would turn the prefixing on, to help avoid collisions
> there it's a fixed prefix, unselectable with a flag in the codec config

This optional prefix will be added following the same pattern as the Beats input. #120 has been opened to track this.